### DISCUSSION TITLE
Adding a Controller to a Collection?

### DISCUSSION THREAD

**Discussion by jgarcia421** [statamic/cms#5621] [Q&A] [answered]
Is it possible to tie a controller to a specific collection? If so, what's the best practice for this? 

Here's my use case: I've got a standard Statamic page. I'm wanting to pull data from a third-party API and then pass that data to the front-end. It seems like it makes sense for this code to live in a Controller, but I'm not sure how to tie that controller in before the page is rendered. It seems like `Statamic::route` almost lets me do this...but I also want to pull in content that's defined within the collection settings in the Statamic control panel. Is there a way to accomplish this?

**Comment by jasonvarga** [comment, marked as answer]
You can just create a route in your regular Laravel routes file, pointing to your own controller.

```php
Route::get('/third-party/{entry}', 'MyController@show');
```
```php
use Statamic\Facades\Entry;
use Statamic\View\View;

public function show($entry)
{
  $entry = Entry::find($entry);

  $collection = $entry->collection();

  $data = [...]; // not sure what you want to do, but you can add whatever variables you need in the view.

  return View::make()
    ->template('templatename')
    ->layout('layoutname')
    ->with($data);
}
```

**Reply by jgarcia421** [reply]
Makes sense. Thanks!
